packages/blocks/src/api/categories.js

```diff
@@ -20,3 +20,13 @@ export function getCategories() {
 export function setCategories( categories ) {
 	dispatch( 'core/blocks' ).setCategories( categories );
 }
+
+/**
+ * Sets the category icon.
+ *
+ * @param {string}       slug Block category slug.
+ * @param {string|Array} icon Block category icon.
+ */
+export function setCategoryIcon( slug, icon ) {
```

Comment:
I'd personally prefer a `addCategory` action instead of this. This actions is targetting the issue directly but I think the issue is not about allowing to edit the icon but more about allowing to set the icon as SVG from the server. I expect some work around this in phase2 (because it's the same issue for blocks).

So instead of offering a way to edit the icon, I'd prefer if we offer a way to add a category without requiring any filter on the backend.

Is this not a viable option?

Reply:
Hi @youknowriad,
I think we made the categories changeable on the server because the blocks may be registered on the server.
I thought we may have timing issues if a block registered on the server, used a category added on the client (the add code may not yet have been executed when the block is registered). So I end up in this approach where the category is still managed on the server but we allow updating the icon with client code so SVG's may be open.
Also, the category seems important data that the server should be aware of, so I was not totally comfortable documenting a method that allows categories being added to the client only. Technically this is already possible using the setCategories action but it is not something we recommend anywhere in our docs.
This approaches documents a way to add categories on the server, and then just the icon is updated on the client. I think updating the icon on the client is fine and could be seen as applying a skin and allows a plugin for example change all the default icons. The icon is not information important to the server while the category may end up being important.

That said I can give a try to an add action make some tests and if things seem fine to update this PR, let me know if you think applying this change is worth it.

Reply:
What if we limit this PR to offering a store action which updates a category on the client instead? What if someone would want to update the name of the category?

```js
const icon = <SVG ....</SVG>;
dispatch( 'core/blocks' ).updateCategory( slug, { icon } );
```

I personally don't see any value in offering all those action wrappers which don't have any additional logic but forces us to have the same functionality documented twice. At the same time, it introduces the confusion for plugin developers because there are 2 ways to perform the same task. I think we had `getCategories` in the past which wasn't using store at all. When we refactored code we didn't remove it which in retrospect should happen in my opinion.

Reply:
Hi @gziolo, I updated this PR to follow your suggestion.